services/src/datasets/external/netcdfcf/mod.rs

```diff
@@ -634,15 +618,12 @@ impl NetCdfCfDataProvider {
         resampling_method: Option<ResamplingMethod>,
         task_context: C,
     ) -> Result<OverviewGeneration> {
-        let mut db_connection = self
-            .metadata_db
-            .get()
-            .await
-            .boxed_context(error::DatabaseConnection)?;
+        let transaction = self.db.write_access(self.id).await?;
```

Comment:
is it a good idea to keep a database transaction open while an overview is created (could take potentially hours)?

Reply:
Good thought (which isn't new for this PR :stuck_out_tongue:).
Pro: It aligns with the overview creation. If this fails, the transaction fails. If it succeeds, it succeeds.
Con: Long-living transactions.

I could manully add another transaction by adding an in-progress flag into the database?